modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/test/CTMessageBusInterceptorTest.java

```diff
@@ -86,64 +109,40 @@ public void setUp() throws Exception {
 		_ctCollection = _ctCollectionLocalService.updateCTCollection(
 			_ctCollection);
 
-		_asyncDestinationServiceRegistration = _registerDestination(
-			_TEST_ASYNC_DESTINATION_NAME,
-			DestinationConfiguration.DESTINATION_TYPE_SERIAL);
-
-		_syncDestinationServiceRegistration = _registerDestination(
-			_TEST_SYNC_DESTINATION_NAME,
-			DestinationConfiguration.DESTINATION_TYPE_SYNCHRONOUS);
-	}
-
-	@After
-	public void tearDown() {
-		_asyncDestinationServiceRegistration.unregister();
-
-		_syncDestinationServiceRegistration.unregister();
+		_testMessageListener.receive(null);
 	}
 
+	@Sync
 	@Test
-	public void testInterceptMessageToAsyncDestination() {
-		Message message = new Message();
+	public void testInterceptMessageToAsyncDestination() throws Exception {
+		long companyId = TestPropsValues.getCompanyId();
 
-		message.setPayload(_TEST_MESSAGE_PAYLOAD);
+		SubscriptionSender subscriptionSender = new SubscriptionSender();
+
+		subscriptionSender.setCompanyId(companyId);
+		subscriptionSender.setMailId(
+			CTMessageBusInterceptorTest.class.getName(), "test");
 
 		try (SafeClosable safeClosable =
 				CTCollectionThreadLocal.setCTCollectionId(
 					_ctCollection.getCtCollectionId())) {
 
-			_messageBus.sendMessage(_TEST_ASYNC_DESTINATION_NAME, message);
-		}
+			_journalArticleLocalService.updateWithUnsafeFunction(
+				journalArticleCTPersistence -> {
+					subscriptionSender.flushNotificationsAsync();
 
-		List<Message> messages = _ctMessageLocalService.getMessages(
-			_ctCollection.getCtCollectionId());
+					return null;
+				});
 
-		Assert.assertSame(messages.toString(), 1, messages.size());
+			Assert.assertNull(_testMessageListener.getReceivedMessage());
 
-		Message deserializedMessage = messages.get(0);
-
-		Assert.assertEquals(
-			_TEST_ASYNC_DESTINATION_NAME,
-			deserializedMessage.getDestinationName());
-		Assert.assertEquals(
-			message.getPayload(), deserializedMessage.getPayload());
+			subscriptionSender.flushNotificationsAsync();
```

Comment:
Hi @Preston-Crary,
I can not understand the logic here. Under change tracker mode, we should always intercept the message, the listener should not receive message at all, why we are asserting listener received message?

Tina

Reply:
You're right.  This was left over from when I had the advice, it was asserting it only captured when called from a CTService.  I'll update this part.